common/src/main/java/org/apache/uniffle/common/provider/HadoopAccessorProvider.java

```diff
@@ -0,0 +1,229 @@
+ * the hadoop cluster is kerberized or not.
+ * <p>
+ * It should be initialized when the shuffle server/coordinator starts. And in client, there is no need
+ * to login with keytab at startup, the authentication of client side should be managed by computing framework
+ * like Spark/MR.
+ */
+public class HadoopAccessorProvider implements Closeable {
+  private static final Logger LOGGER = LoggerFactory.getLogger(HadoopAccessorProvider.class);
+
+  private static volatile HadoopAccessorProvider provider;
+
+  private Map<String, UserGroupInformation> cache = new ConcurrentHashMap<>();
+  private ScheduledExecutorService scheduledExecutorService;
+  private boolean kerberosEnabled = false;
+  private RssBaseConf rssBaseConf;
+  private long reLoginIntervalSec;
+
+  private HadoopAccessorProvider(RssBaseConf rssConf) throws Exception {
+    this.rssBaseConf = rssConf;
+    this.kerberosEnabled = rssConf.getBoolean(RSS_ACCESS_HADOOP_KERBEROS_ENABLE);
+
+    if (kerberosEnabled) {
+      String keytabFile = rssConf.getString(RSS_ACCESS_HADOOP_KERBEROS_KEYTAB_FILE);
+      if (StringUtils.isEmpty(keytabFile)) {
+        throw new Exception("When hadoop kerberos is enabled. the conf of "
+            + RSS_ACCESS_HADOOP_KERBEROS_KEYTAB_FILE.key() + " must be set");
+      }
+
+      String principal = rssConf.getString(RSS_ACCESS_HADOOP_KERBEROS_PRINCIPAL);
+      if (StringUtils.isEmpty(principal)) {
+        throw new Exception("When hadoop kerberos is enabled. the conf of "
+            + RSS_ACCESS_HADOOP_KERBEROS_PRINCIPAL.key() + " must be set");
+      }
+
+      this.reLoginIntervalSec = rssConf.getLong(RSS_ACCESS_HADOOP_KERBEROS_RELOGIN_INTERVAL);
+
+      Configuration conf = new Configuration(false);
+      conf.set("hadoop.security.authentication", "kerberos");
+
+      UserGroupInformation.setConfiguration(conf);
+      UserGroupInformation.loginUserFromKeytab(principal, keytabFile);
+
+      cache = new ConcurrentHashMap<>();
+
+      LOGGER.info("Got Kerberos ticket, keytab [{}], principal [{}], user [{}]",
+          keytabFile, principal, UserGroupInformation.getLoginUser());
+
+      scheduledExecutorService = Executors.newSingleThreadScheduledExecutor(
+          new ThreadFactoryBuilder()
+              .setDaemon(true).setNameFormat("Kerberos-Relogin-%d").build());
+      scheduledExecutorService.scheduleAtFixedRate(
+          HadoopAccessorProvider::kerberosRelogin,
+          reLoginIntervalSec,
+          reLoginIntervalSec,
+          TimeUnit.SECONDS);
+    }
+  }
+
+  @VisibleForTesting
+  static void kerberosRelogin() {
+    try {
+      LOGGER.info("Renewing kerberos token.");
+      UserGroupInformation.getLoginUser().checkTGTAndReloginFromKeytab();
+    } catch (Throwable t) {
+      LOGGER.error("Error in token renewal task: ", t);
+    }
+  }
+
+  private static UserGroupInformation getProxyUser(String user) throws IOException {
+    provider.cache.putIfAbsent(
```

Comment:
When do we clean cache?

Reply:
Only will be cleaned when stopping server/coordinator.